src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs

```diff
@@ -154,7 +154,10 @@ private void OutMember(Expression instance, MemberInfo member, Type exprType)
                 Visit(instance);
                 if (ShouldParenthesisMemberExpression(instance))
                     Out(")");
-
+                
+                if (instance.Type.GetInterfaces().Any(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IDictionary<,>)) && name == "Values")
+                    Out(".ToDictionary(e1 => e1.Key, e1 => e1.Value)");
```

Comment:
Why `e1`? Cannot we have some more "natural" name?

Reply:
Changed

Reply:
Changed to `e2` so it doesn't potentially shadow variable named `x` that user could want to use